pyccel/codegen/compiling/compilers.py

```diff
@@ -441,7 +441,10 @@ def compile_shared_library(self, compile_obj, output_folder, verbose = False, sh
         # Collect compile information
         exec_cmd, includes, libs_flags, libdirs_flags, m_code = \
                 self._get_compile_components(compile_obj, accelerators)
-        linker_libdirs_flags = ['-Wl,-rpath' if l == '-L' else l for l in libdirs_flags]
+        if self._info['exec'] == 'nvcc':
+            linker_libdirs_flags = ['-Xcompiler' if l == '-L' else f'"-Wl,-rpath,{l}"' for l in libdirs_flags]
```

Comment:
This change adds `f'"-Wl,-rpath,{l}"'` where we it wasn't before. Why is this done? I am surprised that this works, but I'm also not sure when the path doesn't start with `-L`

Reply:
https://github.com/pyccel/pyccel-cuda/commit/935e90308bc907a6cc9b1069c67abe57135c802d
I am trying to find the case where it was needed to do it this way.

Reply:
I have just read this properly and remembered that the original list is `["-L", "/path/to/libdir", "-L", "/path/to/libdir", ...]`. So instead of changing this to: `["-Wl,-rpath", "/path/to/libdir", "-Wl,-rpath", "/path/to/libdir", ...]` you are now changing it to: `["-Xcompiler", "-Wl,-rpath,/path/to/libdir", "-Xcompiler", "-Wl,-rpath,/path/to/libdir", ...]`.

This makes more sense now. However I am not sure whether it makes sense to use different ways of specifying the flags depending on compilers.
Also the `-Wl` flag is described as:
> -Wl,option
> Pass option as an option to the linker. If option contains commas, it is split into multiple options at the commas. You can use this syntax to pass an argument to the option. For example, -Wl,-Map,output.map passes -Map output.map to the linker. When using the GNU linker, you can also get the same effect with -Wl,-Map=output.map.

Wouldn't it make more sense therefore to simply do:
`linker_libdirs_flags = ['-Xlinker,-rpath' if l == '-L' else l for l in libdirs_flags]`
?

Reply:
not sure why on my machine `libdirs_flags` always is empty. :thinking:

Reply:
This doesn't compile :(. What about:
```python
linker_key = '-Xlinker' if compiler == 'nvidia' else '-Wl'
linker_libdirs_flags = [linker_key if l == '-L' else f'-rpath,{l}' for l in libdirs_flags]
```
?

Reply:
Can't get the `compiler` from there.
we only have the information about the `vendor` and the `language`

Reply:
[68fea94](https://github.com/pyccel/pyccel-cuda/pull/32/commits/68fea94c714bc60d6b7f78e52fb9eb1052a18d80)

Reply:
In any case my check was about the vendor ^^